src/gennylib/src/PoolFactory.cpp

```diff
@@ -177,7 +177,8 @@ struct PoolFactory::Config {
     };
 };
 
-PoolFactory::PoolFactory(std::string_view rawUri) : _config(std::make_unique<Config>(rawUri)) {}
+PoolFactory::PoolFactory(std::string_view rawUri, mongocxx::options::client& clientOpts)
+    : _config(std::make_unique<Config>(rawUri)), _clientOpts{clientOpts} {}
```

Comment:
if the only thing you actually care about and use from the client options is the apm flag maybe just pass that boolean in? that way it's obvious that only that flag is being respected.

Reply:
I don't think a boolean is sufficient here. The client options are being passed in with the apm options already initialized from the tests. We need a way to pass those options down to the `pool` constructor.